dispatcher/dispatcher.go

```diff
@@ -254,7 +256,7 @@ loop:
 			simplelogger.AssertTrue(d.CurTime <= d.pauseTime)
 			d.goUntilPauseTime()
 
-			if d.ctx.Err() != nil {
+			if d.ctx.Err() != nil || d.stopped {
```

Comment:
There are many instances where 'd.stopped' is checked now, I can maybe try if it also works without that. It was related probably to errors/panic resulting from a dispatcher that's already stopped while the nodes continued sending events, or so.

Reply:
I checked this and indeed at some places the dispatcher simulation loop is still running and trying to do things (e.g. post visualization events) while the related components/channels have already been closed by the exit.  I've now reduced the number of 'd.stopped' checks. Maybe later on a separate PR could be created with a good solution.  (E.g. don't stop any other components until the dispatcher loops are  done running)

Reply:
Gracefully stopping OTNS has been quite a challenge. 

I think current `stopped` is just a flag to avoid calling Stop() method twice.
The dispatcher goroutine actually depends on `d.ctx.Done()` to notify quitting.

Dispatcher should be able to exit processing events when `d.ctx.Done()` is triggered, but I did observe some exiting delays when OTNS simulation is interrupted. The root cause is unclear yet.

Reply:
Tried to reduce the d.stopped checks to a minimum now.  It's still at 2 places, to avoid runtime errors.